File: frontend/src/services/HotelService.js
import api from './ApiService';

class HotelService {
  // Get all hotels
  getAllHotels() {
    return api.get('/hotels');
  }

  // Get hotel by ID
  getHotelById(id) {
    return api.get(`/hotels/${id}`);
  }

  // Search hotels by location
  searchHotelsByLocation(location) {
    return api.get(`/hotels/search?location=${location}`);
  }

  // Search hotels by name
  searchHotelsByName(name) {
    return api.get(`/hotels/search?name=${name}`);
  }
}

export default new HotelService();

File: src/main/java/com/example/bookingtracker/service/HotelService.java
package com.example.bookingtracker.service;

import com.example.bookingtracker.model.Hotel;
import com.example.bookingtracker.repository.HotelRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class HotelService {
    
    private final HotelRepository hotelRepository;
    
    public List<Hotel> getAllHotels() {
        return hotelRepository.findAll();
    }
    
    public Optional<Hotel> getHotelById(Long id) {
        return hotelRepository.findById(id);
    }
    
    public List<Hotel> searchHotelsByLocation(String location) {
        return hotelRepository.findByLocationContainingIgnoreCase(location);
    }
    
    public List<Hotel> searchHotelsByName(String name) {
        return hotelRepository.findByNameContainingIgnoreCase(name);
    }
}